src/main/java/org/cascadebot/cascadebot/music/KaraokeHandler.java

```diff
@@ -0,0 +1,112 @@
+        CascadeBot.INS.getHttpClient().newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(@NotNull Call call, @NotNull IOException e) {
+                Messaging.sendExceptionMessage(channel, Language.i18n(channel.getGuild().getIdLong(), ""), e);
+            }
+
+            @Override
+            public void onResponse(@NotNull Call call, @NotNull Response response) {
+
+                if (response.body() == null) {
+                    Messaging.sendDangerMessage(channel, Language.i18n(channel.getGuild().getIdLong(), "commands.karaoke.cannot_find"));
+                    return;
+                }
+
+                if (response.code() != 200) {
+                    Messaging.sendDangerMessage(channel, Language.i18n(channel.getGuild().getIdLong(), "commands.karaoke.cannot_find"));
+                    return;
+                }
+
+                String body = "";
+                try {
+                    body = response.body().string();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+
+                if (body.equals("")) {
+                    Messaging.sendDangerMessage(channel, Language.i18n(channel.getGuild().getIdLong(), "commands.karaoke.cannot_find"));
+                    return;
+                }
+
+                try {
+                    Document doc = builder.parse(new StringInputStream(body));
+                    NodeList texts = doc.getElementsByTagName("text");
+                    Captions captions = new Captions();
+                    for (int i = 0; i < texts.getLength(); i++) {
+                        NamedNodeMap attrs = texts.item(i).getAttributes();
+                        Node dur = attrs.getNamedItem("dur");
+                        Node start = attrs.getNamedItem("start");
+
+                        if (dur != null && start != null) {
+                            captions.addCaption(StringEscapeUtils.unescapeHtml(texts.item(i).getTextContent()), Double.parseDouble(start.getNodeValue()), Double.parseDouble(dur.getNodeValue()));
+                        }
+                    }
+
+                    while (isKaraoke(guildId)) {
+                        List<String>caption = captions.getCaptions((CascadeBot.INS.getMusicHandler().getPlayer(guildId).getPlayer().getPlayingTrack().getPosition() / 1000D));
+                        EmbedBuilder embed = new EmbedBuilder()
+                                .setDescription(String.join("\n", caption))
+                                .setColor(MessageType.INFO.getColor());
+                        message.editMessage(embed.build()).override(true).queue();
+                        Thread.sleep(14000);
+                    }
+                } catch (IOException | SAXException | MDCException | InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+    }
+
```

Comment:
Hm?

Reply:
Please state the issue here.

Reply:
Yegh I don't know what deadly is pointing out here. Our code style requires a new line at the end of a class so this is correct.

Reply:
GitHub softwrapping messed up again. my bad